hazelcast/src/main/java/com/hazelcast/jet/config/DeltaJobConfig.java

```diff
@@ -0,0 +1,442 @@
+     * {@link Processor} instance in the context of the job.
+     */
+    public Long getMaxProcessorAccumulatedRecords() {
+        return maxProcessorAccumulatedRecords;
+    }
+
+    /**
+     * Sets the maximum number of records that can be accumulated by any single
+     * {@link Processor} instance in the context of the job.
+     * <p>
+     * For more info see {@link InstanceConfig#setMaxProcessorAccumulatedRecords(long)}.
+     * <p>
+     * If set, it has precedence over {@link InstanceConfig}'s one.
+     * <p>
+     * The default value is {@code -1} - in that case {@link InstanceConfig}'s value
+     * is used.
+     */
+    public DeltaJobConfig setMaxProcessorAccumulatedRecords(Long maxProcessorAccumulatedRecords) {
+        checkTrue(maxProcessorAccumulatedRecords == null || maxProcessorAccumulatedRecords > 0
+                        || maxProcessorAccumulatedRecords == -1,
+                "maxProcessorAccumulatedRecords must be a positive number or -1");
+        this.maxProcessorAccumulatedRecords = maxProcessorAccumulatedRecords;
+        return this;
+    }
+
+    /**
+     * Returns maximum execution time for the job in milliseconds.
+     */
+    public Long getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    /**
+     * Sets the maximum execution time for the job in milliseconds. If the
+     * execution time (counted from the time job is submitted), exceeds this
+     * value, the job is forcefully cancelled. The default value is {@code 0},
+     * which denotes no time limit on the execution of the job.
+     */
+    public DeltaJobConfig setTimeoutMillis(Long timeoutMillis) {
+        checkTrue(timeoutMillis == null || timeoutMillis >= 0,
+                "timeoutMillis can't be negative");
+        this.timeoutMillis = timeoutMillis;
+        return this;
+    }
+
+    /**
+     * Returns the configured {@linkplain #setInitialSnapshotName initial
+     * snapshot name} or {@code null} if no initial snapshot is configured.
+     */
+    @Nullable
+    public String getInitialSnapshotName() {
+        return initialSnapshotName;
+    }
+
+    /**
+     * Sets the {@linkplain Job#exportSnapshot exported state snapshot} name to
+     * restore the initial job state from. This state will be used for initial
+     * state and also for the case when the execution restarts before it
+     * produces first snapshot.
+     * <p>
+     * The job will use the state even if {@linkplain
+     * JobConfig#setProcessingGuarantee processing guarantee} is set to {@link
+     * ProcessingGuarantee#NONE}.
+     * <p>
+     * Cannot be used for {@linkplain JetService#newLightJob light jobs}.
+     *
+     * @param initialSnapshotName the snapshot name given to
+     *                            {@link Job#exportSnapshot(String)}
+     * @return {@code this} instance for fluent API
+     */
+    @Nonnull
+    public DeltaJobConfig setInitialSnapshotName(@Nullable String initialSnapshotName) {
```

Comment:
it should not be possible to change `initialSnapshotName`. It was used when the job was submitted. Or do we want to allow to use different snapshot from `Job.resume()`. But if so, this should not be named `initialSnapshotName` to not confuse users.